### PR TITLE
Add podman support in the Jenkins chart

### PR DESCRIPTION
**Pull request by LinuxSuRen**
This is part of https://github.com/kubesphere/ks-devops/issues/67

You can get details from README file.

### CODE REVIEW

**Comment by LinuxSuRen** [comment]
hi @JohnNiang, would you mind helping to review this PR? Thanks.

File: Makefile [hunk 1]
```diff
@@ -56,6 +56,9 @@ render-chart:
 	helm template ks-ctl charts/ks-devops  -n kubesphere-devops-system \
 		--set serviceAccount.create=true --create-namespace \
 		--set image.pullPolicy=Always \
+#		--set jenkins.Agent.Builder.Registry=ghcr.io/linuxsuren \
+#		--set jenkins.Agent.Builder.Maven.Tag=test-podman \
```
Comment (JohnNiang) [inline Makefile:60]: test-podman -> test.

**Review by JohnNiang — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (LinuxSuRen) [reply Makefile:60]: See also https://github.com/LinuxSuRen/devops-agent/pkgs/container/builder-maven

image

**Comment by JohnNiang** [comment]
It should be better If we can install chart by executing command like this:

```bash
make install-chart engine=podman
```

**Comment by LinuxSuRen** [comment]
> It should be better If we can install chart by executing command like this:
> 
> ```shell
> make install-chart engine=podman
> ```

The make command does not support that.

File: Makefile (hunk 1 above)
Comment (LinuxSuRen) [reply Makefile:60]: By the way, the line 59 and 60 is just an example for user if they want to change the image name and tag.

**Comment by JohnNiang** [comment]
If I execute `make render-chart`, a error appears:

```bash
/bin/sh: line 7: --set: command not found
make: *** [Makefile:56: render-chart] Error 127
```

**Comment by LinuxSuRen** [comment]
> make render-chart

It works well in the following environment:

macOS 10
```
helm version
version.BuildInfo{Version:"v3.3.4", GitCommit:"a61ce5633af99708171414353ed49547cf05013d", GitTreeState:"dirty", GoVersion:"go1.15.2"}
```

So, what's the version of helm do you use?

File: Makefile (hunk 1 above)
Comment (JohnNiang) [reply Makefile:60]: I found that we will append `-podman` to the maven tag by default if we choose podman as container runtime.

https://github.com/kubesphere/ks-devops/blob/6024df1d54657f7a3454ba1db83709d2b883fb96/charts/ks-devops/charts/jenkins/templates/_helpers.tpl#L50-L54

https://github.com/kubesphere/ks-devops/blob/6024df1d54657f7a3454ba1db83709d2b883fb96/charts/ks-devops/charts/jenkins/templates/jenkins-casc-config.yml#L108-L111

And I also found [there](https://github.com/LinuxSuRen/devops-agent/pkgs/container/builder-maven) was a image with `test-podman-podman` tag. But I think it is so weird.

**Comment by JohnNiang** [comment]
> > make render-chart
> 
> It works well in the following environment:
> 
> macOS 10
> 
> ```
> helm version
> version.BuildInfo{Version:"v3.3.4", GitCommit:"a61ce5633af99708171414353ed49547cf05013d", GitTreeState:"dirty", GoVersion:"go1.15.2"}
> ```
> 
> So, what version of helm do you use?

image

File: Makefile (hunk 1 above)
Comment (LinuxSuRen) [reply Makefile:60]: That's not a problem. As I mentioned, line 59 and 60 are just the example usage.

**Comment by JohnNiang** [comment]
Hi @LinuxSuRen , I've installed KubeSphere in containerd. But I'm confused how can I test against it.

**Comment by LinuxSuRen** [comment]
> Hi @LinuxSuRen , I've installed KubeSphere on containerd. But I'm confused how can I test against it.

See https://github.com/kubesphere/ks-devops/pull/171/files#diff-17a4ca7a93821a7c1460483d3a9ca5ea9e82f8c554d17f52f382553d2a2ee3fbR3

**Comment by JohnNiang** [comment]
> > Hi @LinuxSuRen , I've installed KubeSphere on containerd. But I'm confused how can I test against it.
> 
> See https://github.com/kubesphere/ks-devops/pull/171/files#diff-17a4ca7a93821a7c1460483d3a9ca5ea9e82f8c554d17f52f382553d2a2ee3fbR3
> 
> After you installed DevOps with the above document instructions. You can create a Pipeline to build your image. You can use the command podman or docker (there is a command alias docker=podman).

In my practical operation, testing this PR is too complicated.

**Comment by LinuxSuRen** [comment]
> In my practical operation, testing this PR is too complicated.

What are your steps? IMO, you just only need to install DevOps with a particular parameter. Then create a Pipeline to run it.

Or, do you have any good suggestion?

**Comment by JohnNiang** [comment]
> > In my practical operation, testing this PR is too complicated.
> 
> What are your steps? IMO, you just only need to install DevOps with a particular parameter. Then create a Pipeline to run it.
> 
> Or, do you have any good suggestions?

It might be better that we provide a ks-installer docker image for testing.

**Comment by LinuxSuRen** [comment]
> It might be better that we provide a ks-installer docker image for testing.

Actually, you don't really need a KubeSphere UI to test it. It only relies on some Jenkins agent images. So you can just install the Jenkins helm chart to test.

**Comment by LinuxSuRen** [comment]
It's pretty easy to install DevOps via helm chart. See also https://github.com/LinuxSuRen/ks-devops/issues/2

But there are some issues that still exist. All those issues do not come from this PR.

**Comment by JohnNiang** [comment]
hi @LinuxSuRen , please rebase upstream.

**Comment by LinuxSuRen** [comment]
> hi @LinuxSuRen , please rebase upstream. There are some changes related with docker registry of ks-devops-apiserver and ks-devops-controller.

Done.

**Comment by JohnNiang** [comment]
### Jenkinsfile to test

```groovy
pipeline {
  agent {
    node {
      label 'base'
    }

  }
  stages {
    stage('stage-q719p') {
      steps {
        container('base') {
          sh '''podman ps
docker ps
docker info
podman info'''
        }

      }
    }

  }
}
```

### Test result

`docker` command dose not work while `podman` is okay.

image

**Comment by LinuxSuRen** [comment]
The alias command does not work currently. In order to make it work well, we need to build a new image. See also https://github.com/kubesphere/devops-agent/blob/fb80a72c6249ef73344d676509075499a19d5427/base/podman/Dockerfile#L62

No matter what, `podman` is our expected.

**Comment by JohnNiang** [comment]
We should be able to ship.

/lgtm

**Pull request merged**